08-Frac_calc/FracCalc.java

```diff
@@ -34,6 +34,11 @@ public static String produceAnswer(String input) {
         op1n = (op1w * op1d) + op1n; //equation for converting into improper fraction
         op2n = (op2w * op2d) + op2n;
 
+        if (op1d == 0 | op2d == 0){
+          String undefined = "Cannot divide by zero";
```

Comment:
Could you not store anything in a local variable and just return directly? This can be better for two reasons: (1) It's easier for the reader to understand - when you store in a local variable the reader can think "Why are they storing this? Are they using it later?" (2) It does not use memory unnecessarily - while in this course this doesn't matter _that_ much (given that what you're storing doesn't take up much space), it's best to get into the habit of not using memory unless you have to.